scrapper.py

```diff
@@ -21,43 +32,189 @@ class IncorrectNumberOfArticlesError(Exception):
+
+                pdf_raw_file.download()
+                text = pdf_raw_file.get_text()
+
+                parts_of_article = text.split('Список литературы')
+
+                self.article.text = ''.join(parts_of_article[:-1])
+                return
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills self.article with meta information
+        """
+
+        self.article.title = article_bs.find('h3').get_text()
+
+        tables = article_bs.find_all('table', {"class": "unntable"})
+
+        for table in tables:
+            table_datas = table.find_all('td')
+
+            if table_datas[0].get_text() != 'Авторы':
+                continue
+
+            link_to_author = table_datas[1].find('a')
+
+            self.article.author = link_to_author.get_text()
+
+        text_date = re.search(r'Поступила в редакцию\s+\d{2}\.\d{2}\.\d{4}', self.article.text)
+
+        if text_date:
+            date_re = re.search(r'\d{2}\.\d{2}\.\d{4}', text_date.group(0))
+
+            self.article.date = datetime.datetime.strptime(date_re.group(0), '%d.%m.%Y')
+
+
 class Crawler:
     """
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self._seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
+        self._article_count = 0
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Finds urls from the given article_bs
+        """
+        table_rows = article_bs.find_all('tr', {"class": "unnrow"})
+
+        for table_row in table_rows:
+            if self._article_count + 1 > self.max_articles:
+                break
+
+            links = table_row.find_all('a')
+
+            for link in links:
+                if self._article_count + 1 > self.max_articles:
+                    break
+
+                match = re.match(r'^\?anum', link['href'])
+
+                if not match:
+                    continue
+
+                self.urls.append(''.join(["http://www.vestnik.unn.ru/ru/nomera", link['href']]))
+                self._article_count += 1
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            if self._article_count + 1 > self.max_articles:
+                break
+
+            response = requests.get(seed_url)
+            article_bs = BeautifulSoup(response.text, features="html.parser")
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self._seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if os.path.isdir(base_path):
+        file_names = os.listdir(base_path)
+
+        for file_name in file_names:
+            os.remove(os.path.join(base_path, file_name))
```

Comment:
the folder should be removed completely. look at shutil module for a better cleanup process